src/components/practice/PracticeUploadModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { QuestionIcon } from '@chakra-ui/icons'
+import {
+  Button,
+  ButtonGroup,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  Tooltip,
+  useDisclosure,
+  useToast,
+} from '@chakra-ui/react'
+import BraftEditor, { EditorState } from 'braft-editor'
+import React, { useState } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+import { defineMessages, useIntl } from 'react-intl'
+import styled from 'styled-components'
+import { commonMessages } from '../../helpers/translation'
+import CommonModal from '../common/CommonModal'
+import FileUploader from '../common/FileUploader'
+import ImageUploader from '../common/ImageUploader'
+
+const messages = defineMessages({
+  practice: { id: 'program.term.practice', defaultMessage: '作業' },
+  uploadByMe: { id: 'program.ui.uploadByMe', defaultMessage: '我要上傳' },
+  uploadPractice: { id: 'program.label.uploadPractice', defaultMessage: '上傳作業' },
+  practiceAttachment: { id: 'program.label.practiceAttachment', defaultMessage: '作品檔案' },
+  practiceAttachmentNotice: { id: 'program.label.practiceAttachmentNotice', defaultMessage: '檔案大小不超過 5GB' },
+  cover: { id: 'program.label.cover', defaultMessage: '封面圖片' },
+  coverNotice: { id: 'program.label.coverNotice', defaultMessage: '建議上傳圖片尺寸為 1600*900px' },
+  description: { id: 'program.label.description', defaultMessage: '描述' },
+  fillTitleNotice: { id: 'program.text.fillTitleNotice', defaultMessage: '請填入標題' },
+  fillDescriptionPlease: { id: 'program.text.fillDescriptionPlease', defaultMessage: '輸入內容描述...' },
+})
+
+const StyledButton = styled(Button)`
+  && {
+    width: 100%;
+  }
+`
+
+const PracticeUploadModal: React.FC = () => {
+  const { formatMessage } = useIntl()
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const toast = useToast()
+  const { register, control, handleSubmit, setError, errors } = useForm<{
+    title: string
+    description?: EditorState
+  }>()
+  const [attachments, setAttachments] = useState<File[]>([])
+  const [coverImage, setCoverImage] = useState<File | null>(null)
+
+  const handleUpload = handleSubmit(({ title, description }) => {
+    if (!title) {
+      setError('title', {
+        message: formatMessage(messages.fillTitleNotice),
+      })
+    }
```

Comment:
NITS: it is better to use `register({ required: 'error message' })` of react-hook-form

```
<Input
  name="title"
  ref={register({ required: formatMessage(messages.fillTitleNotice) })}
  variant="outline"
/>
```